src/Estimators/MagnetizationDensityInput.cpp

```diff
@@ -0,0 +1,119 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2023 QMCPACK developers.
+//
+// File developed by: Raymond Clay, rclay@sandia.gov, Sandia National Laboratories
+//
+// File created by: Raymond Clay, rclay@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+#include "MagnetizationDensityInput.h"
+
+#include <cmath>
+
+#include "OhmmsData/AttributeSet.h"
+#include "Message/UniformCommunicateError.h"
+#include "EstimatorInput.h"
+namespace qmcplusplus
+{
+MagnetizationDensityInput::MagnetizationDensityInput(xmlNodePtr cur)
+{
+  input_section_.readXML(cur);
+  auto setIfInInput = LAMBDA_setIfInInput;
+  setIfInInput(nsamples_, "samples");
+  setIfInInput(integrator_, "integrator");
+  have_center_ = setIfInInput(center_, "center");
+  have_corner_ = setIfInInput(corner_, "corner");
+  have_grid_   = setIfInInput(grid_real_, "grid");
+  have_dr_     = setIfInInput(dr_, "dr");
+}
+
+
+MagnetizationDensityInput::DerivedParameters MagnetizationDensityInput::calculateDerivedParameters(
+    const Lattice& lattice) const
+{
+  PosType corner = 0.0;
+  //Corner and center can be taken care of by defaults.  corner=0 if not specified.
+  if (have_center_)
+    corner = center_ - lattice.Center;
+  else if (have_corner_)
+    corner = corner_;
+
+  TinyVector<int, DIM> grid;
+
+  //dr or grid must be specified to perform the grid math.  Input already checked before we get here.  
+  if (have_dr_)
+    for (int d = 0; d < DIM; ++d)
+    {
+      grid[d] = (int)std::ceil(std::sqrt(dot(lattice.Rv[d], lattice.Rv[d])) / dr_[d]);
+      //Now we know the lattice and can compute grid points from dr.  We check it's validity here.
+      if (grid[d] < 1)
+        throw UniformCommunicateError(
```

Comment:
This can happen I think only for a negative dr value and the `checkParticularValidity` means this can never happen.  Am I wrong?  If I'm not I'd just drop the check, and add a comment to this function that it has a precondition that all `dr_  > 0`

Reply:
I agree with that.  I'll remove that redundant check.